### ISSUE TITLE
align not equivalent to ==

### ISSUE THREAD

**Issue by mergesort** [robb/Cartography#87]
I'm getting different results for two layouts, which I believe should be equivalent.

The first one works, and produces views on the screen.

```
constrain(messageButton, emailButton, twitterButton) { messageButton, emailButton, twitterButton in
    if let superview = messageButton.superview {
        messageButton.top == superview.top
        emailButton.top == messageButton.top
        twitterButton.top == emailButton.top

        messageButton.bottom == superview.bottom
        emailButton.bottom == messageButton.bottom
        twitterButton.bottom == emailButton.bottom
    }
}
```

The second one which I believe should be equivalent doesn't work.

```
constrain(messageButton, emailButton, twitterButton) { messageButton, emailButton, twitterButton in
    if let superview = messageButton.superview {
        align(top: messageButton, emailButton, twitterButton, superview)
        align(bottom: messageButton, emailButton, twitterButton, superview)
    }
}
```

With an output of:
```
(
    "<NSLayoutConstraint:0x7ff4b6516f00 UIButton:0x7ff4b65001b0.top == UIButton:0x7ff4b6500390.top>",
    "<NSLayoutConstraint:0x7ff4b6517040 UIButton:0x7ff4b6500390.top == UIButton:0x7ff4b6503740.top>",
    "<NSLayoutConstraint:0x7ff4b65184e0 V:|-(0)-[UIButton:0x7ff4b65001b0]   (Names: '|':[redacted].ShareView:0x7ff4b6509910 )>",
    "<NSLayoutConstraint:0x7ff4b65186e0 UIButton:0x7ff4b65001b0.bottom == [redacted].ShareView:0x7ff4b6509910.bottom>",
    "<NSLayoutConstraint:0x7ff4b6417070 V:[[redacted].ShareView:0x7ff4b6509910(64)]>",
    "<NSAutoresizingMaskLayoutConstraint:0x7ff4b6526770 h=--& v=--& UIButton:0x7ff4b6503740.midY ==>",
    "<NSLayoutConstraint:0x7ff4b651af40 UIButton:0x7ff4b65001b0.bottom == UIButton:0x7ff4b6503740.bottom>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x7ff4b651af40 UIButton:0x7ff4b65001b0.bottom == UIButton:0x7ff4b6503740.bottom>
```

**Comment by robb** [comment]
@mergesort I went ahead and merged #88, lemme know if this fixes this one and then I'll cut `0.3`. :sparkles:

**Comment by mergesort** [comment]
All good. \o/
